rca/utils/models.py

```diff
@@ -373,6 +379,84 @@ def get_context(self, request, *args, **kwargs):
         return context
 
 
+class LegacySiteTag(TagBase):
+    class Meta:
+        verbose_name = "legacy site tag"
+        verbose_name_plural = "legacy site tags"
+
+
+class LegacySiteTaggedPage(ItemBase):
+    tag = models.ForeignKey(
+        LegacySiteTag, related_name="tagged_pages", on_delete=models.CASCADE
+    )
+    content_object = ParentalKey(
+        to=Page, on_delete=models.CASCADE, related_name="tagged_items"
+    )
+
+
+class LegacyNewsAndEventsMixin(models.Model):
+    legacy_news_and_event_tags = ClusterTaggableManager(
+        verbose_name="Legacy news and event tags",
+        help_text=(
+            "Specify one or more tags to identify related news and events from "
+            "the legacy site. A maximum of three items with the same combination "
+            "of tags will then be displayed on the page."
+        ),
+        through=LegacySiteTaggedPage,
+        blank=True,
+    )
+
+    class Meta:
+        abstract = True
+
+    @property
+    def legacy_news_cache_key(self):
```

Comment:
should this be `legacy_news_and_events_cache_key`? Seen as though it's news _and_ events that will be in the cache. Guess it's only called internally anyway so not a big deal

Reply:
yes, it just seemed to be getting a bit too length/repetitive.